packages/gatsby/src/utils/tracer/jaeger-local.ts

```diff
@@ -0,0 +1,37 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { initTracer, JaegerTracer, TracingConfig } from "jaeger-client"
+
+let tracer: JaegerTracer
+
+function create(): JaegerTracer {
+  // See schema
+  // https://github.com/jaegertracing/jaeger-client-node/blob/master/src/configuration.js#L37
+  const config: TracingConfig = {
+    serviceName: `gatsby`,
+    reporter: {
+      // Provide the traces endpoint; this forces the client to
+      // connect directly to the Collector and send spans over HTTP
+      collectorEndpoint: `http://localhost:14268/api/traces`,
+    },
+    sampler: {
+      type: `const`,
+      param: 1,
+    },
+  }
+  const options = {}
+  tracer = initTracer(config, options)
+  return tracer
+}
+
+function stop(): Promise<void> {
+  return new Promise(resolve => {
+    // It's not on the @types definition, but the Jaeger documentation tells us to call
+    // close(): https://github.com/jaegertracing/jaeger-client-node#trace-buffer
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
```

Comment:
We can add it ourselves using .d.ts files - which could be temporary - and we could PR `@types/jaeger-client` to add it there

Reply:
I found there is PR open already - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44132 - so we should track it

Reply:
Rather than ts-ignore, I'd add the type inline: 
```typescript
(tracer as any & {close(callback?: () => void): void;).close(resolve)
```
(Not checked)

Reply:
Since there's a PR already open, it probably makes sense for me to watch it and then update this PR when it's merged.

Reply:
I'm fine with waiting - can we mark the PR as draft in meantime then? And if the PR I linked won't be handled in reasonable time (~1-2 weeks) - we could use trick that @ascorbic shown so this PR doesn't get stale

Reply:
Changed to draft, and added an extended interface and comment which can be removed once the other PR is merged.

Reply:
I think we can resolve this comment thread and mark as ready to review again (from draft) as PR was merged and published in DefenitelyTyped